src/faultlog/unresolved_pels.cpp

```diff
@@ -0,0 +1,223 @@
+
+using Objects = std::map<sdbusplus::message::object_path, Interfaces>;
+
+constexpr auto STATE_CONFIGURED = "CONFIGURED";
+constexpr auto STATE_DECONFIGURED = "DECONFIGURED";
+
+struct GuardedTarget
+{
+    pdbg_target* target = nullptr;
+    std::string phyDevPath = {};
+    GuardedTarget(const std::string& path) : phyDevPath(path)
+    {}
+};
+
+static int getGuardedTarget(struct pdbg_target* target, void* priv)
+{
+    // recursive callback function that exits when the target matching the
+    // guarded targets physical path is found in the device tree.
+    // to exit the recusive function return 1 to continue return 0
+    GuardedTarget* guardTarget = reinterpret_cast<GuardedTarget*>(priv);
+    ATTR_PHYS_DEV_PATH_Type phyPath;
+    if (!DT_GET_PROP(ATTR_PHYS_DEV_PATH, target, phyPath))
+    {
+        if (strcmp(phyPath, guardTarget->phyDevPath.c_str()) == 0)
+        {
+            guardTarget->target = target;
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void UnresolvedPELs::populate(sdbusplus::bus::bus& bus,
+                              GuardRecords& guardRecords,
+                              std::vector<uint32_t>& processedEid,
+                              json& jsonNag)
+{
+    log<level::INFO>("UnresolvedPELs::populate");
+    using Objects = std::map<sdbusplus::message::object_path, Interfaces>;
+
+    Objects objects;
+    auto method = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "org.freedesktop.DBus.ObjectManager", "GetManagedObjects");
+    auto reply = bus.call(method);
+    reply.read(objects);
+    for (auto& iter : objects)
+    {
+        bool resolved = true;
+        uint32_t id = 0;
+        for (auto& intf : iter.second)
+        {
+            if (intf.first != "xyz.openbmc_project.Logging.Entry")
+            {
+                continue;
+            }
+            for (auto& propertyMap : intf.second)
+            {
+                if (propertyMap.first == "Id")
+                {
+                    auto idPtr = std::get_if<uint32_t>(&propertyMap.second);
+                    if (idPtr == nullptr)
+                    {
+                        continue;
+                    }
+                    id = *idPtr;
+                }
+                else if (propertyMap.first == "Resolved")
+                {
+                    auto resolvedPtr = std::get_if<bool>(&propertyMap.second);
+                    if (resolvedPtr == nullptr)
+                    {
+                        continue;
+                    }
+                    resolved = *resolvedPtr;
+                }
+            }
+            if (resolved)
+            {
+                continue;
+            }
+
+            // get pel json file
+            std::string pel;
+            auto method2 = bus.new_method_call(
+                "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+                "org.open_power.Logging.PEL", "GetPELJSON");
```

Comment:
GetPELJSON can be pretty slow on hostboot PELs because their python plugins can be slow to parse all of their  UserData sections.  There can be up to 3000 PELs, and it being a few seconds per PEL can really add up...

I guess since this only runs every now and then, maybe it isn't a big deal, but at  the very least you'll want to skip informational errors, since hostboot has a lot of them but you won't ever care about them.

Ideally we'd add a new D-Bus method on our org.open_power.Logging.PEL interface to return just the data you want.

Reply:
service engineer makes the decision to change the reolved bit i guess, not by any application

Reply:
> GetPELJSON can be pretty slow on hostboot PELs because their python plugins can be slow to parse all of their UserData sections. There can be up to 3000 PELs, and it being a few seconds per PEL can really add up...
> 

Yes I had noticed it is takin a second or 2 for every pel, probably a new interface to get the required data would be ideal.

Reply:
It may be a good idea to run the idea of the CE/SSR doing the resolving by Carl to make sure he'd be OK with it.

Reply:
> It may be a good idea to run the idea of the CE/SSR doing the resolving by Carl to make sure he'd be OK with it.

sure will check with them